ResearchKit/Common/ORKDiscreteGraphView.m

```diff
@@ -0,0 +1,140 @@
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+*/
+
+ 
+#import "ORKDiscreteGraphView.h"
+#import "ORKGraphView_Internal.h"
+#import "ORKHelpers.h"
+#import "ORKAxisView.h"
+#import "ORKCircleView.h"
+
+
+@implementation ORKDiscreteGraphView
+
+#pragma mark - Init
+
+- (void)sharedInit {
+    [super sharedInit];
+    self.shouldConnectRanges = YES;
+}
+
+#pragma mark - Draw
+
+- (BOOL)shouldDrawLinesForPlotIndex:(NSInteger)plotIndex {
+    return [self numberOfValidValues] > 0 && self.shouldConnectRanges;
+}
+
+- (CAShapeLayer *)plotLineLayerForPlotIndex:(NSInteger)plotIndex withPath:(CGPathRef)path {
+    CAShapeLayer *layer = [super plotLineLayerForPlotIndex:plotIndex withPath:path];
+    layer.lineWidth = 8.0;
+    return layer;
+}
+
+- (void)drawLinesForPlotIndex:(NSInteger)plotIndex {
+    
+    CGFloat positionOnXAxis = CGFLOAT_MAX;
+    ORKRangePoint *positionOnYAxis = nil;
+    
+    for (NSUInteger i=0; i<self.yAxisPoints.count; i++) {
+        
+        ORKRangePoint *dataPointVal = self.dataPoints[i];
+        
+        if (!dataPointVal.isEmpty && !dataPointVal.isRangeZero) {
+            
+            UIBezierPath *plotLinePath = [UIBezierPath bezierPath];
+            
+            positionOnXAxis = [self.xAxisPoints[i] floatValue];
+            positionOnXAxis += [self offsetForPlotIndex:plotIndex];
+            positionOnYAxis = ((ORKRangePoint *)self.yAxisPoints[i]);
+            
+            [plotLinePath moveToPoint:CGPointMake(positionOnXAxis, positionOnYAxis.minimumValue)];
+            [plotLinePath addLineToPoint:CGPointMake(positionOnXAxis, positionOnYAxis.maximumValue)];
+            
+            CAShapeLayer *plotLineLayer = [self plotLineLayerForPlotIndex:plotIndex withPath:plotLinePath.CGPath];
+            
+            [self.plotsView.layer addSublayer:plotLineLayer];
+            [self.pathLines addObject:plotLineLayer];
+        }
+    }
+}
+
+- (CGFloat)offsetForPlotIndex:(NSInteger)plotIndex {
+    CGFloat pointWidth = 8.0;
+    
+    NSInteger numberOfPlots = [self numberOfPlots];
+    
+    CGFloat offset = 0;
+    
+    if (numberOfPlots%2 == 0) {
+        //Even
+        offset = (plotIndex - numberOfPlots/2 + 0.5) * pointWidth;
+    } else {
+        //Odd
+        offset = (plotIndex - numberOfPlots/2) * pointWidth;
+    }
+    
+    return offset;
+}
+
+#pragma mark - Graph Calculations
+
+- (CGFloat)canvasYPointForXPosition:(CGFloat)xPosition {
+    BOOL snapped = [self.xAxisPoints containsObject:@(xPosition)];
+    CGFloat canvasYPosition = 0;
+    if (snapped) {
+        NSInteger positionIndex = [self yAxisPositionIndexForXPosition:xPosition];
+        canvasYPosition = ((ORKRangePoint *)self.yAxisPoints[positionIndex]).maximumValue;
+    }
+    return canvasYPosition;
+}
+
+#pragma mark -- Animation
+
+- (void)updateScrubberViewForXPosition:(CGFloat)xPosition {
+    CGFloat scrubbingVal = [self valueForCanvasXPosition:(xPosition)];
+    if (scrubbingVal == NSNotFound) {
```

Comment:
Have some concerns over comparing a `CGFloat` with a `NSInteger` (`NSNotFound`). Noticing some other places in this patch use CGFLOAT_MAX as an invalid value, possibly we can do 
`#define ORK_CGFLOAT_INVALID CGFLOAT_MAX`,
then use `ORK_CGFLOAT_INVALID` to replace `CGFLOAT_MAX` and `NSNotFound` as a predefined invalid value.